From 22ed442796ff34d619af63d110896c2e383d317d Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Mon, 6 Jan 2020 10:10:57 -0600
Subject: [PATCH 37/43] arm64: dts: logicpd-imx8mm-kit: Fix ILI2117 sleep

With the custom driver removed, and the driver from 5.5 backported,
this updates the device tree to point to the new driver.

It also enables a pull-down resistor to help prevent floating
inputs.

Fixes: PHOENIX8-26

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/lpd-imx8mm-lvds.dtsi | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/lpd-imx8mm-lvds.dtsi b/arch/arm64/boot/dts/freescale/lpd-imx8mm-lvds.dtsi
index 2913f31f2da4..c76d6caed120 100644
--- a/arch/arm64/boot/dts/freescale/lpd-imx8mm-lvds.dtsi
+++ b/arch/arm64/boot/dts/freescale/lpd-imx8mm-lvds.dtsi
@@ -41,14 +41,11 @@
 
 &i2c2 {
 	ili_touch: ilitouch@26 {
-		compatible = "ili,ili2117a";
+		compatible = "ilitek,ili2117";
 		reg = <0x26>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_touchscreen>;
-		interrupts-extended = <&gpio1 9 IRQ_TYPE_LEVEL_LOW>;
-		ili2117a,poll-period = <20>;
-		ili2117a,max-touch = <2>;
-		wakeup-source;
+		interrupts-extended = <&gpio1 9 IRQ_TYPE_EDGE_RISING>;
 	};
 
 	dsi_lvds_bridge: sn65dsi84@2d {
@@ -140,7 +137,7 @@
 
 		pinctrl_touchscreen: touchscreengrp {
 			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO09_GPIO1_IO9		0x19
+				MX8MM_IOMUXC_GPIO1_IO09_GPIO1_IO9		0x119
 			>;
 		};
 	};
-- 
2.17.1

